aten/src/ATen/native/cuda/AveragePool2d.cu

```diff
@@ -21,7 +23,11 @@ __device__ inline int max(int a, int b) {
 }
 
 template <typename scalar_t, typename accscalar_t>
-__global__ void avg_pool2d_out_cuda_frame(const int nthreads,
+__global__ void
+#if __CUDA_ARCH__ == 620
+  __launch_bounds__(CUDA_NUM_THREADS, 1)
```

Comment:
In the <<<>>> launches below in the file, kernels are not launched with `CUDA_NUM_THREADS`, they're launched with 
```
const uint32_t num_threads = std::min(at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, 1024);
```
Therefore, the largest the block could be is 1024, and you should use
 ```
  __launch_bounds__(CUDA_NUM_THREADS, 1)
  ```

Reply:
should be appropriate `C10_LAUNCH_BOUNDS` everywhere

Reply:
editing